File: src/org/jetbrains/agda/core/CFunctionDeclaration.java
package org.jetbrains.agda.core;

import java.util.ArrayList;
import java.util.List;

/**
 * @author Evgeny.Kurbatsky
 */
public class CFunctionDeclaration extends CDeclaration {
    private String myName;
    private CExpression myType;
    private List<FunctionBody> myBodes = new ArrayList<FunctionBody>();

    public CFunctionDeclaration(String name, CExpression type) {
        this.myName = name;
        this.myType = type;
    }

    public String getName() {
        return myName;
    }

    public CExpression getType() {
        return myType;
    }

    public List<FunctionBody> getBodyes() {
        return myBodes;
    }

    @Override
    public String toString() {
        StringBuilder builder = new StringBuilder();
        builder.append(myName).append(" : ").append(myType).append(" {\n");

        for (FunctionBody expression : myBodes) {
            builder.append("\t").append(expression.toString()).append("\n");
        }

        builder.append("}");
        return builder.toString();
    }
}

File: src/org/jetbrains/agda/core/CFunctionDeclaration.kt
package org.jetbrains.agda.core

import java.util.ArrayList

/**
 * @author Evgeny.Kurbatsky
 */
public class CFunctionDeclaration(val name : String, aType : CExpression, val bodies : List<CFunctionBody>) : CDeclaration(aType) {

    public fun toString() : String {
        val builder : StringBuilder = StringBuilder()
        builder.append(name).append(" : ").append(aType).append(" {\n")
        for (expression in bodies)
        {
            builder.append("\t").append(expression.toString()).append("\n")
        }
        builder.append("}")
        return builder.toString()
    }
}
